mockld/event_data.go

```diff
@@ -17,41 +13,10 @@ type Event ldvalue.Value
 // Events is an array of events. This specialized type provides helper methods.
 type Events []Event
 
-// EventUser is a JSON representation of a user in an event. LaunchDarkly's JSON schema for users
-// in events is slightly different than users as SDK inputs.
-type EventUser ldvalue.Value
```

Comment:
There's no longer a good reason to have this specialized type, so I've removed a lot of related stuff to simplify things. I was using it mostly for two reasons: 1. in functions like `EventIsFeatureEvent` that took user properties as a parameter, where `lduser.User` wouldn't have been right because that is the input state of the user and not the _event_ representation of a user, and 2. to provide the "canonicalize" logic which was only needed because I was doing equality tests on the whole event.